asset-transfer-private-data/application-javascript/app.js

```diff
@@ -0,0 +1,262 @@
+
+    // register & enroll application user with CA, which is used as client identify to make chaincode calls, stores app user identity in local wallet
+    await caUtils.RegisterOrgUser(caUtils.Org2UserId, mspOrg2, walletOrg2, caService);
+
+    try {
+        // Create a new gateway for connecting to Org's peer node.
+        const gatewayOrg2 = new Gateway();        
+        await gatewayOrg2.connect(ccpOrg2,
+            { wallet: walletOrg2, identity: caUtils.Org2UserId, discovery: { enabled: true, asLocalhost: true } });
+        
+        return gatewayOrg2;
+    } catch (error) {
+        console.error(`Error in connecting to gateway: ${error}`);
+        process.exit(1);
+    }
+}
+
+// Main workflow : usecase details at asset-transfer-private-data/chaincode-go/README.md
+// This app uses fabric-samples/test-network based setup and the companion chaincode
+// For this usecase illustration, we will use both Org1 & Org2 client identity from this same app
+// In real world the Org1 & Org2 identity will be used in different apps to achieve asset transfer.
+async function main() {
+    try {       
+
+        /** ******* Fabric client init: Using Org1 identity to Org1 Peer ********** */
+        const gatewayOrg1 = await initContractFromOrg1Identity();
+        const networkOrg1 = await gatewayOrg1.getNetwork(myChannel);
+        const contractOrg1 = networkOrg1.getContract(myChaincodeName);
+
+        /** ~~~~~~~ Fabric client init: Using Org2 identity to Org2 Peer ~~~~~~~ */
+        const gatewayOrg2 = await initContractFromOrg2Identity();
+        const networkOrg2 = await gatewayOrg2.getNetwork(myChannel);
+        const contractOrg2 = networkOrg2.getContract(myChaincodeName);        
+
+        try {           
+            // Sample transactions are listed below
+            // Add few sample Assets & transfers one of the asset from Org1 to Org2 as the new owner
+            let assetID1 = "asset1";
+            let assetID2 = "asset2";
+            const assetType = "ValuableAsset";
+            let result;
+            let asset1Data = {objectType: assetType, assetID: assetID1, color: "green", size: 20,  appraisedValue: 100};
+            let asset2Data = {objectType: assetType, assetID: assetID2, color: "blue", size: 35,  appraisedValue: 727};
+
+            console.log('\n**************** As Org1 Client ****************');            
+            console.log('Adding Assets to work with: Submit Transaction: CreateAsset '+assetID1);
+            let statefulTxn = contractOrg1.createTransaction('CreateAsset');
+            //this chaincode function has a check to ensure that client from same org can send Txn to the org's peer
+            statefulTxn.setEndorsingOrganizations(mspOrg1);
+            let tmapData = Buffer.from(JSON.stringify(asset1Data));
+            statefulTxn.setTransient( {
+                asset_properties: tmapData
+            });
+            result = await statefulTxn.submit();                        
```

Comment:
seeing this log in client , every other run or so , but the txn **seem to go through fine**. 
```2020-07-22T21:19:53.701Z - error: [DiscoveryHandler]: Error: Endorsement plan group does not contain enough peers (0) to satisfy policy (required:1)```

Note: Using `setEndorsingOrganizations(mspOrg1)` in txn. Policy used `"OR('Org1MSP.peer','Org2MSP.peer')"`

From the chaincode logs it seem to have worked in committing to ledger, and subsequent reads are seeing the data written from above call.

It was puzzling me, Any pointers?